protos/flyteidl/service/plugin_system.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+package flyteidl.service;
+
+option go_package = "github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/service";
+import "flyteidl/core/literals.proto";
+import "flyteidl/core/tasks.proto";
+import "flyteidl/core/interface.proto";
+
+// BackendPluginService defines an RPC Service that allows propeller to send the request to the backend plugin server.
+service BackendPluginService {
+  // Send a task create request to the backend plugin server.
+  rpc CreateTask (TaskCreateRequest) returns (TaskCreateResponse){};
+  // Get job status.
+  rpc GetTask (TaskGetRequest) returns (TaskGetResponse){};
+  // Delete the task resource.
+  rpc DeleteTask (TaskDeleteRequest) returns (TaskDeleteResponse){};
+}
+
+// The state of the execution is used to control its visibility in the UI/CLI.
+enum State {
+  FAILED = 0;
+  PENDING = 1;
+  RUNNING = 2;
+  SUCCEEDED = 3;
+}
+
+// Represents a request structure to create task.
+message TaskCreateRequest {
+  // The inputs required to start the execution. All required inputs must be
+  // included in this map. If not required and not provided, defaults apply.
+  // +optional
+  core.LiteralMap inputs = 1;
+  // Template of the task that encapsulates all the metadata of the task.
+  core.TaskTemplate template = 2;
+  // Prefix for where offloaded data from user workflows will be written.
+  string output_prefix = 3;
```

Comment:
added it

Reply:
Sorry, I meant what is an example of the data that is offloaded. I not fully understanding where this will be used. Or is this the tasks output data rather than the offloaded data?

Reply:
yup, this the tasks output. For example, we will pass it to databricks job, and the spark task will write the dataframe to this output_prefix.

Reply:
OK, now I get it. The description should be updated to reflect this. I think the phrase "offloaded data from user workflows" is a little difficult to understand. This is the location that task should write it's output data I think.